### PR TITLE
Feature 5 fitness report

### PR DESCRIPTION
**Pull request by dennymin**
final project feature 5 fitness report

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -128,4 +128,30 @@ app.get('/api/your/meals', (req, res, next) => {
...
+         "workouts"."length",
+         "workouts"."caloriesBurned",
+         DATE("workouts"."date"),
+         STRING_AGG(("muscleGroup"."name"), ', ') as "muscles"
+  from   "workouts"
+  join   "workoutMuscleGroups" using ("workoutId")
+  join   "muscleGroup" using ("muscleId")
+  where  "userId" = 1
+  and    DATE("workouts"."date") >= '${startDate}'
+  and    DATE("workouts"."date") <= '${endDate}'
```
Comment (thebearingedge) [inline server/index.js]: We never put untrusted data into our SQL. Use a parameterized query instead.

File: client/pages/your-workouts.jsx [hunk 2]
```diff
@@ -76,11 +76,9 @@ export default function YourWorkouts(props) {
...
         setExpanded(!expanded);
       };
-      const workoutDate = new Date(workout.date);
-      const workingDate = workoutDate.toDateString().split('');
-      workingDate.splice(3, 0, ',');
-      workingDate.splice(11, 0, ',');
-      const renderedDate = workingDate.join('');
+      const placeholder = workout.date.slice(0, 10);
+      const workoutDate = new Date(placeholder);
+      const workingDate = workoutDate.toGMTString().slice(0, 16);
```
Comment (thebearingedge) [inline client/pages/your-workouts.jsx]: I recommend using the `date-fns` library for all of your date formatting.

File: client/pages/home.jsx [hunk 3]
```diff
@@ -1,9 +1,232 @@
...
+    if (serverData[i].muscles.includes('Back')) {
+      returningNumbers.musclesWorked.Back = returningNumbers.musclesWorked.Chest + 1;
+    }
+    if (serverData[i].muscles.includes('Arms')) {
+      returningNumbers.musclesWorked.Arms = returningNumbers.musclesWorked.Chest + 1;
+    }
+    if (serverData[i].muscles.includes('Legs')) {
+      returningNumbers.musclesWorked.Legs = returningNumbers.musclesWorked.Chest + 1;
+    }
+  }
```
Comment (thebearingedge) [inline client/pages/home.jsx]: This is _a lot_ of logic to be re-run every time the component renders. Instead, format the data on the server into the correct shape before sending it to the client.

File: client/pages/home.jsx [hunk 4]
```diff
@@ -1,9 +1,232 @@
...
+  const queryString = '?' + params.toString();
+
+  useEffect(() => {
+    let isCanceled = false;
+    const serverAddress = '/api/your/fitness' + queryString;
+    fetch(serverAddress)
+      .then(response => response.json())
+      .then(data => {
+        !isCanceled && pullServerData(data);
+      }, [null, startDate, endDate]);
```
Comment (thebearingedge) [inline client/pages/home.jsx]: I think you mean to put this array on line 73.

File: client/pages/home.jsx [hunk 5]
```diff
@@ -1,9 +1,232 @@
...
+  const defaultStartYear = lastWeekDate.getFullYear();
+  let defaultStartMonth = `${lastWeekDate.getMonth() + 1}`;
+  if (defaultStartMonth.length === 1) {
+    defaultStartMonth = '0' + defaultStartMonth;
+  }
+  let defaultStartDay = `${lastWeekDate.getDate()}`;
+  if (defaultStartDay.length === 1) {
+    defaultStartDay = '0' + defaultStartDay;
+  }
+  const defaultStartPicker = `${defaultStartYear}-${defaultStartMonth}-${defaultStartDay}`;
```
Comment (thebearingedge) [inline client/pages/home.jsx]: I recommend `date-fns` for all of your date math/formatting.

File: client/app.jsx [hunk 6]
```diff
@@ -54,7 +54,10 @@ export default class App extends React.Component {
   }
 
   render() {
-    const title = this.state.route.path.split('/').splice(1, 2).join(' ').toUpperCase();
+    let title = this.state.route.path.split('/').splice(1, 2).join(' ').toUpperCase();
+    if (title === '') {
+      title = 'FITNESS REPORT';
+    }
```
Comment (thebearingedge) [inline client/app.jsx]: It would probably make more sense if each page just had a header in it with a hard-coded title.

**Review by thebearingedge — CHANGES_REQUESTED** [review]

**Comment by dennymin** [comment]
final project feature 5 fitness report correction

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**